From a7bee66ed22f54556509156a05273bf5268e5a5d Mon Sep 17 00:00:00 2001
From: Cheick TRAORE <cheick.traore-ext@st.com>
Date: Fri, 19 Apr 2024 14:26:59 +0200
Subject: [PATCH] arm64: dts: st: Add lpuart1 in the stm32mp257f-ev1 board

Add lpuart1 node in stm32mp257f-ev1.dts and in the serial aliases.

Change-Id: If09a798e6cf03bddea92fe02ea468bd8ad079160
Signed-off-by: Cheick TRAORE <cheick.traore-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/375110
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
index a3424658c239..a429e6218f71 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -25,6 +25,7 @@ aliases {
 		ethernet1 = &eth1;
 		serial0 = &usart2;
 		serial1 = &usart6;
+		serial2 = &lpuart1;
 	};
 
 	chosen {
@@ -459,6 +460,17 @@ timer {
 	};
 };
 
+&lpuart1 {
+	pinctrl-names = "default", "idle", "sleep";
+	pinctrl-0 = <&lpuart1_pins_a>;
+	pinctrl-1 = <&lpuart1_idle_pins_a>;
+	pinctrl-2 = <&lpuart1_sleep_pins_a>;
+	/delete-property/dmas;
+	/delete-property/dma-names;
+	uart-has-rtscts;
+	status = "disabled";
+};
+
 &ltdc {
 	default-on;
 	rotation-memory = <&ltdc_sec_rotation>;
-- 
2.39.5

